src/main/java/com/adaptavist/tm4j/jenkins/extensions/buildsteps/FeatureFilesDownloader.java

```
@@ -53,27 +58,28 @@ public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskLi
         List<JiraInstance> jiraInstances = getDescriptor().getJiraInstances();
         String path = workspace.getRemote() + "/";
         try {
-            if (isEmpty(this.projectKey)) {
-                throw new RuntimeException(PROJECT_KEY_IS_REQUIRED);
-            }
+        	new Validator().validateProjectKey(this.projectKey)
+        		.validateTargetPath(this.targetPath)
+        		.serverAddress(this.serverAddress);
             String tql = format("testCase.projectKey = '%s'", this.projectKey);
             Tm4jJiraRestClient tm4jJiraRestClient = new Tm4jJiraRestClient(jiraInstances, serverAddress);
             tm4jJiraRestClient.exportFeatureFiles(getFeatureFilePath(path), tql, logger);
         } catch (NoTestCasesFoundException e) {
             logger.printf("%s No feature files found. %n", ERROR);
```

Should we throw an exception here as well to stop the build?
```
throw new RuntimeException(e);
```